python/cudf/cudf/core/column/column.py

```diff
@@ -321,22 +322,22 @@ def _fill(
         if end <= begin or begin >= self.size:
             return self if inplace else self.copy()
 
-        fill_scalar = as_device_scalar(fill_value, self.dtype)
+        device_value = as_device_scalar(fill_value, self.dtype)
```

Comment:
if possible we want to try and use a `cudf.Scalar` in the python layer and then get the `DeviceScalar` via `.device_value` in cython right before it hits libcudf. This should avoid any unnecessary synchronization from interacting with the scalar here, such as when calling `is_valid()`.

Reply:
Thanks! I'm sure this is why I used `cudf.Scalar` before.